k8s/upgrades/0.8.0-0.8.1/cstor_volume_upgrade.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+set -e
+
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+
+function usage() {
+    echo 
+    echo "Usage:"
+    echo 
+    echo "$0 <pv-name> <openebs-namespace>"
+    echo 
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <openebs-namespace> Get the namespace where openebs"
+    echo "  pods are installed"
+    exit 1
+}
+
+function setDeploymentRecreateStrategy() {
+    dns=$1 # deployment namespace
+    dn=$2  # deployment name
+    currStrategy=`kubectl get deploy -n $dns $dn -o jsonpath="{.spec.strategy.type}"`
+
+    if [ $currStrategy = "RollingUpdate" ]; then
+       kubectl patch deployment --namespace $dns --type json $dn -p "$(cat patch-strategy-recreate.json)"
+       rc=$?; if [ $rc -ne 0 ]; then echo "Upgrade failed | ERROR: $rc"; exit; fi
+       echo "Deployment upgrade strategy set as recreate"
+    else
+       echo "Deployment upgrade strategy was already set as recreate"
+    fi
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+pv=$1
+ns=$2
+
+# Check if pv exists
+kubectl get pv $pv &>/dev/null;check_pv=$?
+if [ $check_pv -ne 0 ]; then
+    echo "$pv not found";exit 1;    
+fi
+
+# Check if CASType is cstor
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.annotations.openebs\.io/cas-type}"`
+if [ $cas_type != "cstor" ]; then
+    echo "Cstor volume not found";exit 1;
+fi
+
+pvc=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.name}"`
+sc_ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+sc_name=`kubectl get pv $pv -o jsonpath="{.spec.storageClassName}"`
+sc_res_ver=`kubectl get sc $sc_name -n $sc_ns -o jsonpath="{.metadata.resourceVersion}"`
+
+################################################################# 
+# STEP: Generate deploy, replicaset and container names from PV #
+#                                                               #
+# NOTES: Ex: If PV="pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc",  #
+#                                                               #
+# c-dep: pvc-cec8e86d-0bcc-11e8-be1c-000c298ff5fc-target        # 
+#################################################################
+
+c_dep=$(echo $pv-target)
```

Comment:
instead of this, there are ways to get the deployment, svc names like
```
for deployment: target_dep=$(kubectl get deploy -n $ns \
>  -l openebs.io/persistent-volume=$pv \
>  -o jsonpath="{range .items[*]}{@.metadata.name}{end}")
for rs: kubectl get rs -n openebs  -l openebs.io/persistent-volume=$pv  -o jsonpath="{range .items[?(@.metadata.ownerReferences[0].name=='$target_dep')]}{@.metadata.name}{end}"
for svc: target_svc=$(kubectl get service -n $ns \
 -l openebs.io/persistent-volume=$pv \
 -o jsonpath="{range .items[*]}{@.metadata.name}{end}")
```